### PR TITLE
add propertynames(QRSparse)

### PR DESCRIPTION
**Pull request by KlausC**
Add `propertynames` compatible with existing `getproperty` for `QRSparse` in `stdlib/SuiteSparse`.

### CODE REVIEW

File: stdlib/SuiteSparse/src/spqr.jl [hunk 1]
```diff
@@ -312,6 +312,11 @@ julia> F.pcol
     end
 end
 
+function Base.propertynames(F::QRSparse, private::Bool=false)
+    public = (:R, :Q, :prow, :pcol)
+    (public..., (private ? setdiff(fieldnames(typeof(F)), public) : ())...)
```
Comment (stevengj) [inline stdlib/SuiteSparse/src/spqr.jl]: Can the compiler infer the type of the result of this?

Comment (KlausC) [reply stdlib/SuiteSparse/src/spqr.jl]: `@code_warntype` says no (infer type is `Any`).
Is it really a requirement? Do you have a suggestion, how to achieve that?
The complier cannot infer the return type in the following example either:
```
julia> function foo2(F, private::Bool=false)
           p = private ? (:A, :B, :C, :D) : (:A, :B)
       end
foo2 (generic function with 2 methods)

julia> @code_warntype foo2(qrf)
Body::Tuple{Symbol,Symbol}
2 1 ─      goto 2 if not false                                                            │╻ foo2
  2 ┄ %2 = π ((:A, :B), Tuple{Symbol,Symbol})                                             ││
  └──      :(goto 3)                                                                      ││
  3 ─      return %2                                                                      │ 

julia> @code_warntype foo2(qrf, true)
Body::Union{Tuple{Symbol,Symbol}, NTuple{4,Symbol}}
2 1 ─      goto 3 if not %%private                                                              │
  2 ─      :(goto 4)                                                                            │
  3 ─      nothing                                                                              │
  4 ┄ %4 = φ (2 => (:A, :B, :C, :D), 3 => (:A, :B))::Union{Tuple{Symbol,Symbol}, NTuple{4,Symbol}}
  └──      return %4                                                                            │
```
But even `fieldnames(typeof(qrf))` is not clean. Also the corresponding implementsation of `propertynames` for `LinearAlgebra.QRCompactWY{Float64,Array{Float64,2}}` is not better.

Comment (stevengj) [reply stdlib/SuiteSparse/src/spqr.jl]: The `foo2` inference seems better than `Any`...

But I guess it's not a big deal for this function, which is mainly used for REPL tab completion.

**Comment by KlausC** [comment]
The previous commit now infers the same type, which `fieldnames` does (that is `Tuple`).

**Review by stevengj — APPROVED** [review]

**Pull request merged**